src/tree/v2/BranchDataItem.ts

```diff
@@ -50,8 +51,10 @@ export class BranchDataItem implements ResourceGroupsItem, Wrapper {
     type: string;
 }
 
-export type BranchDataItemFactory = (branchItem: ResourceModelBase, branchDataProvider: BranchDataProvider<ApplicationResource, ResourceModelBase>, options?: BranchDataItemOptions) => BranchDataItem;
+export type BranchDataItemFactory = (branchItem: ResourceModelBase, branchDataProvider: BranchDataProvider<ApplicationResource, ResourceModelBase>, options?: BranchDataItemOptions) => BranchDataItem | CompatibleBranchDataItem;
```

Comment:
IMO we should make `CompatibleBranchDataItem` extend `BranchDataItem` instead of making this `BranchDataItem | CompatibleBranchDataItem`